src/query/service/tests/it/storages/system.rs

```diff
@@ -203,6 +205,14 @@ async fn test_credits_table() -> Result<()> {
     Ok(())
 }
 
+async fn test_catalogs_table(file: &mut impl Write) -> Result<()> {
```

Comment:
Please also add a stateless test.

Reply:
we do not need stateless test for system tables. It's hard to use.  It's easy to cover in golden file tests.